resources/sdk/purecloudjavascript/templates/ApiClient.mustache

```diff
@@ -1163,6 +1194,61 @@ class ApiClient {
 		return url;
 	}
 
+    /**
+    	 * Builds http client.
+    	 * @returns {String} The http client default or the one that is injected.
+    	 */
+    	buildHttpClient() {
+    		if (this.httpClient) {
+    		    return this.httpClient;
+    		} else {
+    		    return this.defaultClient();
+    		}
+    	}
+
+        buildRequestOptions(url, method, headers, data, params) {
+             var request = new RequestOptions();
+             request.setMethod(method);
+             request.setUrl(url);
+             if (data) {
+                 request.setData(data);
+             }
+
+             if (params) {
+               request.setParams(params);
+             }
+
+             request.headers =  headers;
+             return request;
+        }
+
+         defaultClient() {
+                // Initialize proxyAgent as null
+                let proxyAgent = null;
+
+                // If agentOptions are provided, create a new https.Agent with those options
+                if (this.agentOptions) {
+                    proxyAgent = new https.Agent(this.agentOptions);
+                }
+
+                // If that.proxyAgent is available, use it (prioritize this over agentOptions)
+                if (this.proxyAgent) {
+                    proxyAgent = this.proxyAgent;
+                }
+
+                var httpClient =  new DefaultHttpClient();
```

Comment:
It could be good to avoid doing unnecessary new on DefaultHttpClient. I mean doing it once only.

Reply:
done